docs/valhalla_elevation_service.md

```diff
@@ -0,0 +1,66 @@
+
+## Valhalla Elevation Service API Reference
+
+Valhalla also consists of a free, open-source elevation service that lets you access digital elevation model data which is useful in computing steepness of edges in the route graph or generating an elevation profile along a computed route.  This page documents the inputs and outputs to the service.
+
+The Valhalla elevation service is in active development. You can follow the [Mapzen blog](https://mapzen.com/blog) to get updates. To report software issues or suggest enhancements, open an issue in GitHub (use the [Thor repository](https://github.com/valhalla/skadi). You can also send a message to routing@mapzen.com.
+
+#### API keys and Service Limits
+
+To use the Valhalla elevation service, you must first obtain a free developer API key from Mapzen. Sign in at https://mapzen.com/developers to create and manage your API keys.
+
+Valhalla consists of a free, shared elevation service. As such, there are limitations on the number of locations to prevent individual users from degrading the overall system performance.
+
+Limits may be increased in the future, but you can contact routing@mapzen.com if you encounter rate limit status messages and need higher limits in the meantime.
+
+### Inputs for the Elevation Service
+
+We currently have two types of actions that can be requested from the Elevation Service.  The first is a profile request that can be used to generate a graph along a computed route since this request returns an array of x & y at each shape point.  The second request is an elevation request that can be used to compute the steepness of edges since the request returns an array of height.  Both requests takes the form of `valhalla.dev.mapzen.com/<profile|elevation>?json={}&api_key=`, where the JSON inputs inside the ``{}`` include location information (shape or an encoded polyline).
```

Comment:
Might mention that x is generally range or distance along the path and y is the elevation or height. Also, the second request is mostly useful to get the height at a specific location (not the steepness which really is more easily computed from the first type of request since it reguires distance between locations in order to return steepness/gradient.